jujugui/static/gui/src/app/components/deployment-flow/deployment-flow.js

```diff
@@ -625,42 +661,52 @@ YUI.add('deployment-flow', function() {
       @returns {Object} The markup.
     */
     _generateAgreementsSection: function() {
-      var status = this._getSectionStatus('agreements');
+      const status = this._getSectionStatus('agreements');
       if (!status.visible) {
         return;
       }
-      var disabled = this.props.acl.isReadOnly();
+      const disabled = this.props.acl.isReadOnly();
       return (
-        <div>
-          <div className="deployment-flow__deploy-option">
-            <input className="deployment-flow__deploy-checkbox"
-              disabled={disabled}
-              id="emails"
-              type="checkbox" />
-            <label className="deployment-flow__deploy-label"
-              htmlFor="emails">
-              Please email me updates regarding feature
-              announcements, performance suggestions, feedback
-              surveys and special offers.
-            </label>
-          </div>
-          <div className="deployment-flow__deploy-option">
-            <input className="deployment-flow__deploy-checkbox"
-              disabled={disabled}
-              id="terms"
-              type="checkbox" />
-            <label className="deployment-flow__deploy-label"
-              htmlFor="terms">
-              I agree that my use of any services and related APIs
-              is subject to my compliance with the applicable&nbsp;
-              <a href="" target="_blank">Terms of service</a>.
-            </label>
-          </div>
+        <div className="deployment-flow__deploy-option">
+          <input className="deployment-flow__deploy-checkbox"
+            onChange={this._handleTermsAgreement}
+            disabled={disabled}
+            id="terms"
+            type="checkbox" />
+          <label className="deployment-flow__deploy-label"
+            htmlFor="terms">
+            I agree to all terms.
+          </label>
         </div>);
     },
 
+    /**
+      Handle change events when the user clicks on the "I agree to terms"
+      checkbox.
+
+      @method _handleTermsAgreement
+      @param {Object} e The change event.
+    */
+    _handleTermsAgreement: function(e) {
+      this.setState({termsAgreed: e.target.checked});
+    },
+
     render: function() {
-      const disabled = this.props.acl.isReadOnly() || this.state.deploying;
+      // If terms are not yet agreed, check to see if terms are visible. If
+      // they are, then they are not yet agreed to.
+      const termsVisible = this._getSectionStatus('agreements').visible;
+      let termsAgreed = this.state.termsAgreed;
+      termsAgreed = termsAgreed === null ? !termsVisible : termsAgreed;
```

Comment:
What? Woah, I had to read this carefully. Is there a way to make this more expressive?
Also, in what cases termsAgreed can be null?